api/ownership.go

```diff
@@ -37,6 +37,10 @@ func OwnershipSetUsernameFromContext(ctx context.Context, srcOwnership *Ownershi
 	// Check if the context has information about the user. If not,
 	// then security is not enabled.
 	if userinfo, ok := auth.NewUserInfoFromContext(ctx); ok {
+		// Public users cannot provide ownership
+		if userinfo.Public {
```

Comment:
This probably should be an accessor func

Reply:
Added UserInfo.IsPublic()